pkg/resource/generator.go

```diff
@@ -104,9 +104,7 @@ func (g *Generator) syncStorage(cr *regopapi.Config, modified *bool) error {
 		return err
 	}
 
-	if len(driver.GetStorageName()) == 0 {
```

Comment:
I'm pretty sure that this was causing syncStorage to create a new bucket after the finalizer ran